docker-compose.yml

```diff
@@ -198,12 +204,17 @@ services:
       - PCTASKS_COSMOSDB__KEY=${PCTASKS_COSMOSDB__KEY:-C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw==}
       # Set trigger app setting separately to avoid issues with __ in env var names
       - FUNC_COSMOSDB_CONN_STR=AccountEndpoint=${PCTASKS_COSMOSDB__URL:-https://172.16.238.246:8081/};AccountKey=${PCTASKS_COSMOSDB__KEY:-C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw==};
+      - FUNC_STORAGE_ACCOUNT_CONN_STR=DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://azurite:10000/devstoreaccount1;QueueEndpoint=http://azurite:10001/devstoreaccount1;TableEndpoint=http://azurite:10002/devstoreaccount1;
+      - FUNC_STORAGE_QUEUE_ACCOUNT_URL=http://azurite:10001/devstoreaccount1
+      - FUNC_STORAGE_ACCOUNT_NAME=devstoreaccount1
+      - FUNC_STORAGE_ACCOUNT_KEY=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==
       # Isolate Change Feed functions for unit tests.
       - PCTASKS_COSMOSDB__TEST_CONTAINER_SUFFIX=${PCTASKS_COSMOSDB__TEST_CONTAINER_SUFFIX-}
       # Used in the various function.json to dynmically set the Cosmos DB container to watch.
       - FUNC_WORKFLOWS_COLLECTION_NAME=workflows${PCTASKS_COSMOSDB__TEST_CONTAINER_SUFFIX-}
       - FUNC_WORKFLOW_RUNS_COLLECTION_NAME=workflow-runs${PCTASKS_COSMOSDB__TEST_CONTAINER_SUFFIX-}
-
+      - FUNC_STORAGE_EVENTS_COLLECTION_NAME=storage-events${PCTASKS_COSMOSDB__TEST_CONTAINER_SUFFIX-}
```

Comment:
Getting additionally bit by PCTASKS_COSMOSDB__TEST_CONTAINER_SUFFIX. When running unit tests, in particular `pctasks/run/tests/workflow/test_remote.py::test_remote_processes_job_with_two_tasks` (and a few others), I get a timeout error like ` Exception: Timeout while waiting for workflow e7a510c85fda49069000d4c79e2b33cf`. That's with the test environment having the PCTASKS_COSMOSDB__TEST_CONTAINER_SUFFIX, and the servers not having the PCTASKS_COSMOSDB__TEST_CONTAINER_SUFFIX set. I then ran this again after doing an `export PCTASKS_COSMOSDB__TEST_CONTAINER_SUFFIX=test` in the console I ran `scripts/servers` in. This fixed the pctasks.run unit tests, but the Azure Function containers failed to come up as part of the docker-compose, because the containers didn't exist.

Ideally there's a way that has unit tests and dev infra run with as little intervention as possible, but still allows the unit tests to have their own temp containers to test the functions.

Reply:
> Are we requiring this now as an environment variable to be set in the dev environment?

I thought that the `storage-events${PCTASKS_COSMOSDB__TEST_CONTAINER_SUFFIX-}` syntax defaulted to `storage-events` if the environment variable is unset. Do you know if there's a way to do that?

> Ideally there's a way that has unit tests and dev infra run with as little intervention as possible, but still allows the unit tests to have their own temp containers to test the functions.

Any suggestions on how to achieve that? It seems to me like we *need*

1. The Cosmos containers to exist *before* the Azure Functions start.
2. The Azure Functions must start before the tests (maybe this could be relaxed? But it would be difficult).

So I would recommend that `temp_cosmosdb_if_emulator` be changed to not delete containers between test sessions. We'd still get test containers for each session used by everything (which preserves the main improvements of #207, of not using the "real" Cosmos containers from the functions). Locally, you might run into issues with state being preserved between runs, but again that's no worse than before #207.

Reply:
>I thought that the storage-events${PCTASKS_COSMOSDB__TEST_CONTAINER_SUFFIX-} syntax defaulted to storage-events if the environment variable is unset. Do you know if there's a way to do that?

In the case where the Azure Functions failed to start, PCTASKS_COSMOSDB__TEST_CONTAINER_SUFFIX was set in order to avoid the other errors I was facing.

The `temp_cosmosdb_if_emulator` was originally intended to create and then delete temporary containers for unit tests. That behavior changed to fail if the PCTASKS_COSMOSDB__TEST_CONTAINER_SUFFIX isn't set in the environment running the unit test code. So there's three types of errors I see:
-  PCTASKS_COSMOSDB__TEST_CONTAINER_SUFFIX isn't set in the unit test environment; temp_cosmosdb_if_emulator fails when it tries to delete tmp containers
- PCTASKS_COSMOSDB__TEST_CONTAINER_SUFFIX is set in the test environment, but not in the server environment. Tests that monitor the containers for RunRecords fail because it's waiting to see records in a tmp container, but the server is writing to the non-tmp containers.
- PCTASKS_COSMOSDB__TEST_CONTAINER_SUFFIX is set in the test environment and the server environment. In this case, the Azure Functions fail to start because the tmp containers don't exist.
- I'm unclear on the behavior where PCTASKS_COSMOSDB__TEST_CONTAINER_SUFFIX  is set in both environments, and those containers exist. I would assume that would solve the Azure Function startup issue, but wouldn't the `temp_cosmosdb_if_emulator` function end up deleting those containers during the unit tests? Or because they get recreated things pass?

Maybe this is a way forward:
- separate out the logic of `temp_cosmosdb_if_emulator`, which could revert back to its original processing (creating and deleting test containers). In that case it wouldn't use PCTASKS_COSMOSDB__TEST_CONTAINER_SUFFIX
- PCTASKS_COSMOSDB__TEST_CONTAINER_SUFFIX is only used for integration tests. In unit tests where the server is interacted with, e.g. `pctasks/run/tests/workflow/test_remote.py`, the default containers are used (as they were before)
- Integration tests require some additional setup, which creates the test containers, sets the PCTASKS_COSMOSDB__TEST_CONTAINER_SUFFIX env, and runs the tests. Perhaps the script should do the actual cleanup of deleting the containers. This could happen in scripts/test-integration

If unit tests and integration tests use separate containers, then the Azure Functions would have be brought down and back up between integration tests and unit tests, where PCTASKS_COSMOSDB__TEST_CONTAINER_SUFFIX is set only for the unit tests. This could also be encapsulated in scripts/test-integration.

The above is a suggestion; more complex than I think is ideal, but the complexity could be pushed into `scripts/test-integration` and makes the environment for running integration tests more complex than unit tests, which seems like the correct ordering.